### DISCUSSION TITLE
Weak formulation of the biharmonic equation

### DISCUSSION THREAD

**Discussion by zoharl3** [kinnala/scikit-fem#1126] [Q&A] [answered]
Maybe you can say a few words on the choice of bilinear form in the biharmonic examples:

https://github.com/kinnala/scikit-fem/blob/9.1.1/docs/examples/ex02.py

https://github.com/kinnala/scikit-fem/blob/9.1.1/docs/examples/ex20.py

It's an alternative to the more common one:

https://math.stackexchange.com/questions/4914421/weak-formulation-of-the-biharmonic-equation

My guess is that it's because it behaves better due to mixture of derivatives that don't vanish on the chosen elements:
- For a Morley element, the Laplacian bilinear form mostly vanishes, and the stiffness matrix is singular.
- For the Argyris, the Laplacian bilinear form result is ugly.

In comparison, the result of the Hessian bilinear form looks good in both cases.

**Comment by kinnala** [comment, marked as answer]
Suppose you are solving the Kirchhoff plate bending problem and want to use other boundary conditions than fully clamped. Then you will get incorrect natural boundary conditions applied if "the more common" formulation is used.

**Reply by zoharl3** [reply]
Okay, I see a related comment in ex02.py.
Thanks
